content/projects/gsoc/2021/projects/cloudevents-plugin.adoc

```diff
@@ -28,15 +28,82 @@ As the CI/CD world is moving more towards interoperability between multiple plat
 Jenkins currently does not support cloudevents, making it hard for users to use it with other platforms which support them.
 
 ==== Project Details
-This project idea proposes to implement a Jenkins plugin which extends the Jenkins to make cloudevents both discoverable and subscribable. Users should be able to set a global configuration to allow users to subscribe and discover cloudevents.
+This project idea proposes to implement a Jenkins plugin which extends Jenkins to make cloudevents both discoverable and subscribable. Users should be able to set a global configuration to allow users to subscribe and discover cloudevents.
 
 The project requires the student to start with plugin development from scratch and then work with understanding how to use the Cloudevents Java SDK to create and read events.
 
 In the beginning, the student can work on understanding the Jenkins plugin development by writing code for creating a simple Build Step which creates a CloudEvent, and from there we can move to the GlobalPluginConfiguration on how Jenkins should listen to the cloudevents.
 
+=== Implementation
+
+===== Jenkins as a Source
```

Comment:
ASCIIDoc wants the heading depth to increase by only 1.  This will fix the CI build
```suggestion
==== Jenkins as a Source
```